zboxcore/sdk/chunked_upload_blobber.go

```diff
@@ -341,49 +346,40 @@ func (sb *ChunkedUploadBlobber) processWriteMarker(
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		logger.Logger.Error("Ref path: Resp", err)
-		return nil, nil, 0, err
+		return nil, nil, 0, nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
-		return nil, nil, 0, fmt.Errorf("Reference path error response: Status: %d - %s ", resp.StatusCode, string(body))
+		return nil, nil, 0, nil, fmt.Errorf("Reference path error response: Status: %d - %s ", resp.StatusCode, string(body))
 	}
 
 	err = json.Unmarshal(body, &lR)
 	if err != nil {
 		logger.Logger.Error("Reference path json decode error: ", err)
-		return nil, nil, 0, err
+		return nil, nil, 0, nil, err
 	}
 
-	//process the commit request for the blobber here
+	rootRef, err := lR.GetDirTree(su.allocationObj.ID)
 	if err != nil {
-
-		return nil, nil, 0, err
+		return nil, nil, 0, nil, err
 	}
-	rootRef, err := lR.GetDirTree(su.allocationObj.ID)
-	if lR.LatestWM != nil {
 
+	if lR.LatestWM != nil {
 		rootRef.CalculateHash()
 		prevAllocationRoot := encryption.Hash(rootRef.Hash + ":" + strconv.FormatInt(lR.LatestWM.Timestamp, 10))
-		// TODO: it is a concurrent change conflict on database.  check concurrent write for allocation
 		if prevAllocationRoot != lR.LatestWM.AllocationRoot {
 			logger.Logger.Info("Allocation root from latest writemarker mismatch. Expected: " + prevAllocationRoot + " got: " + lR.LatestWM.AllocationRoot)
```

Comment:
Shouldn't we return error here?

Reply:
Actually yes. We should merge commitworker and chunked upload blobber's commit request, prob. in next PR.
I returned with error in commitworker.go for same issue but missed here.